packages/cli/Cargo.toml

```diff
@@ -88,6 +88,76 @@ prettyplease = { workspace = true }
+const-serialize = { workspace = true, features = ["serde"] }
+
+# Image compression/conversion
+# - JPEG
+mozjpeg = { version = "0.10.7", default-features = false, features = [
+    "parallel",
+] }
+# - PNG
+imagequant = "4.2.0"
+png = "0.17.9"
+# Image format/conversion
+image = { version = "0.25", features = ["avif"] }
+
+# CSS Minification
+lightningcss = "1.0.0-alpha.60"
+
+# Js minification - swc has introduces minor versions with breaking changes in the past so we pin all of their crates
+swc = "=0.283.0"
+swc_allocator = { version = "=0.1.8", default-features = false }
+swc_atoms = { version = "=0.6.7", default-features = false }
+swc_cached = { version = "=0.3.20", default-features = false }
+swc_common = { version = "=0.37.5", default-features = false }
+swc_compiler_base = { version = "=0.19.0", default-features = false }
+swc_config = { version = "=0.1.15", default-features = false }
+swc_config_macro = { version = "=0.1.4", default-features = false }
+swc_ecma_ast = { version = "=0.118.2", default-features = false }
+swc_ecma_codegen = { version = "=0.155.1", default-features = false }
+swc_ecma_codegen_macros = { version = "=0.7.7", default-features = false }
+swc_ecma_compat_bugfixes = { version = "=0.12.0", default-features = false }
+swc_ecma_compat_common = { version = "=0.11.0", default-features = false }
+swc_ecma_compat_es2015 = { version = "=0.12.0", default-features = false }
+swc_ecma_compat_es2016 = { version = "=0.12.0", default-features = false }
+swc_ecma_compat_es2017 = { version = "=0.12.0", default-features = false }
+swc_ecma_compat_es2018 = { version = "=0.12.0", default-features = false }
+swc_ecma_compat_es2019 = { version = "=0.12.0", default-features = false }
+swc_ecma_compat_es2020 = { version = "=0.12.0", default-features = false }
+swc_ecma_compat_es2021 = { version = "=0.12.0", default-features = false }
+swc_ecma_compat_es2022 = { version = "=0.12.0", default-features = false }
+swc_ecma_compat_es3 = { version = "=0.12.0", default-features = false }
+swc_ecma_ext_transforms = { version = "=0.120.0", default-features = false }
+swc_ecma_lints = { version = "=0.100.0", default-features = false }
+swc_ecma_loader = { version = "=0.49.1", default-features = false }
+swc_ecma_minifier = { version = "=0.204.0", default-features = false }
+swc_ecma_parser = { version = "=0.149.1", default-features = false }
+swc_ecma_preset_env = { version = "=0.217.0", default-features = false, features = [
+    "serde",
+] }
+swc_ecma_transforms = { version = "=0.239.0", default-features = false }
+swc_ecma_transforms_base = { version = "=0.145.0", default-features = false }
+swc_ecma_transforms_classes = { version = "=0.134.0", default-features = false }
+swc_ecma_transforms_compat = { version = "=0.171.0", default-features = false }
+swc_ecma_transforms_macros = { version = "=0.5.5", default-features = false }
+swc_ecma_transforms_module = { version = "=0.190.0", default-features = false }
```

Comment:
it really would not be terrible to cut out the asset system (Manifest, optimizer, etc) into its own crate again... and maybe with some feature flags so we can gate `dev` mode `dx` to not have to include all this.

Reply:
Agreed. I would like to pull out asset optimization and resolution into a separate crate, but I don't want to block the release and it would take longer to make an more stable API. How about a follow up PR after more pressing issues like the base path are fixed? I have an issue ready once this is merged